File: 2014-12-01/solution.rb
#!/usr/bin/env ruby

def getData()
    File.read('./data/pg47498.txt')
end

def wordCount()
    book = getData
    words = {}
    ordered_words = []

    book.downcase.scan(/\w+/).each do |word|
        words[word] = 0 if words[word] == nil
        words[word] += 1
    end

    words.each do |word, count|
        ordered_words.push [word, count]
    end

    ordered_words.sort! do |a,b|
        b[1] <=> a[1]
    end

    10.times do |index|
        puts ordered_words[index][0] + ': ' + ordered_words[index][1].to_s
    end
end

wordCount

File: 2014-12-01/solution.js
(function(){

    'use strict';

    var fs = require('fs');


    function main() {
        var words = {};
        var file = fs.readFileSync('./data/pg47498.txt', {encoding: 'utf8'});

        file = file.toLowerCase()
                    .replace(/[^a-zA-Z0-9|\s]/g, '')
                    .replace(/[\s]/g, ' ')
                    .split(' ');


        file.forEach(function(word){
            if (words[word] === undefined) {
                words[word] = 0;
            }

            words[word] += 1;
        });
    }

    main();

}());
